tree/20_cfe_basics/ncf_lib.cf

```diff
@@ -1265,3 +1309,75 @@ body replace_with ncf_add_parameter_with_separator_and_enclosing_characters(para
  replace_value => "${match.1}${match.2}${match.3}${parameter_separator}${parameter}${match.4}";
 }
 
+bundle agent slackware_boot_services(service, action, old_class_prefix, class_prefix)
+{
+  vars:
+    "canon_service" string => canonify("${service}");
+
+  classes:
+    
+    "action_enable" expression => strcmp("enable", ${action});
+    "action_disable" expression => strcmp("disable", ${action});
+    "action_is_enabled" expression => strcmp("is-enabled", ${action});
+
+  methods:
+    
+    action_disable::
+      "chmod 644" usebundle => perm_on_file("/etc/rc.d/rc.${service}", "644", "${old_class_prefix}", "${class_prefix}");
+
+    action_enable::
+      "chmod 755" usebundle => perm_on_file("/etc/rc.d/rc.${service}", "755", "${old_class_prefix}", "${class_prefix}");
+      "block in init scripts" usebundle => service_block_rc_d("${service}", "${old_class_prefix}", "${class_prefix}");
+
+    action_is_enabled::
+      "enable dry run" usebundle => push_dry_run_mode("true");
+      "chmod 755" usebundle => perm_on_file("/etc/rc.d/rc.${service}", "755", "${canon_service}_init_script_execmod", "${canon_service}_init_script_execmod");
+      "block in init scripts" usebundle => service_block_rc_d("${service}", "${old_class_prefix}", "${class_prefix}");
```

Comment:
I can add it to force the bundle to stop checking on failing the first condition (755 mod on the init script).
It does work without it however, because in case it fails and then meets the "block in script" requirement, the _not_ok classes will still get reported.